pymanopt/manifolds/psd.py

```diff
@@ -76,12 +77,8 @@ def ehess2rhess(self, x, egrad, ehess, u):
                 multisym(multiprod(multiprod(u, multisym(egrad)), x)))
 
     def norm(self, x, u):
-        # This implementation is as fast as np.linalg.solve_triangular and is
-        # more stable, as the above solver tends to output non positive
-        # definite results.
-        c = la.cholesky(x)
-        c_inv = la.inv(c)
-        return la.norm(multiprod(multiprod(c_inv, u), multitransp(c_inv)))
+        xinvu = la.solve(x, u)
+        return np.sqrt(np.tensordot(xinvu, multitransp(xinvu), axes=x.ndim))
```

Comment:
This effectively reimplements `inner` (as is often the case). I'd therefore suggest to change inner to something like this
```python
    def inner(self, x, u, v):
        xinvu = la.solve(x, u)
        if u is v:
            xinvv = xinvu
        else:
            xinvv = la.solve(x, v)
        return np.tensordot(xinvu, multitransp(xinvv), axes=x.ndim)
```
and change `norm` to
```suggestion
        return np.sqrt(self.inner(x, u, v))
```